packages/devtools/src/components/ActivitiesGrid/ActivitiesGrid.tsx

```diff
@@ -0,0 +1,150 @@
+};
+
+function filterActivities(event: ActivityEvent, params: URLSearchParams): boolean {
+  for (const [key, filter] of params.entries()) {
+    const value = getPath(
+      {
+        ...event,
+        path: getActivityPath(event.body),
+      },
+      key
+    );
+
+    if (value != filter) {
+      return false;
+    }
+  }
+  return true;
+}
+
+function getColumnClassName(columnId: ColumnId): `column${Capitalize<ColumnId>}` {
+  return columnClassMap[columnId];
+}
+
+interface ActivitiesGridProps {
+  list: ActivityEvent[];
+  selected: ActivityEvent | undefined;
+  setSelected: (event: ActivityEvent | undefined) => void;
+  params: URLSearchParams;
+  setParams: (params: URLSearchParams) => void;
+}
+
+const ActivitiesGrid: FC<ActivitiesGridProps> = ({
+  list,
+  selected,
+  setSelected,
+  params,
+  setParams,
+}) => {
+  const classes = useActivitiesGridClasses();
+  const activityPaths = [...new Set(list.map((event) => getActivityPath(event.body)))].sort();
+
+  const handleRowSelect = (event: ActivityEvent) => {
+    if (selected && selected.id === event.id) {
+      setSelected(undefined);
+    } else {
+      setSelected(event);
+    }
+  };
+
+  const handleTypeFilter = (path: string) => {
+    if (params.get('path') === path) {
+      params.delete('path');
+    } else {
+      params.set('path', path);
+    }
+    setParams(params);
+  };
+
+  return (
+    <div className={classes.tableContainer}>
+      <table className={classes.table}>
+        <thead className={classes.tableHeader}>
+          <tr>
+            <th scope="col" className={mergeClasses(classes.headerCell, classes.columnType)}>
+              <Menu hasIcons>
+                <MenuTrigger disableButtonEnhancement>
+                  <MenuButton className={classes.menuButton}>
+                    <span className={classes.typeLabel}>Type</span>
+                    {activityPaths.length > 0 && <Filter20Regular />}
+                  </MenuButton>
+                </MenuTrigger>
+                <MenuPopover className={classes.menuPopover}>
+                  <MenuList>
+                    {activityPaths.map((path) => (
+                      <MenuItem
+                        key={path}
+                        onClick={() => handleTypeFilter(path)}
+                        icon={params.has('path', path) ? <CheckmarkFilled /> : <></>}
+                      >
+                        {path}
+                      </MenuItem>
+                    ))}
+                  </MenuList>
+                </MenuPopover>
+              </Menu>
+            </th>
+            {GRID_COLUMNS.slice(1).map((column) => (
+              <th
+                key={column.id}
+                scope="col"
+                className={mergeClasses(classes.headerCell, classes[getColumnClassName(column.id)])}
+              >
+                {column.label}
+              </th>
+            ))}
+          </tr>
+        </thead>
+        <tbody>
+          {list.length ? (
```

Comment:
not important here because the app is pretty small, but fwiw this kind of stuff is what react says memoization (or the react compiler) is supposed to help with. this is in theory quite expensive as the list grows. unless you're noticing any choppy render fps drops (particularly when there are hundreds or thousands of events in the list), I wouldn't sweat it, but something to keep an eye on.